resources/assets/js/validations.js

```diff
@@ -410,100 +380,68 @@ $.extend(true, laravelValidation, {
+        DateFormat: function(value, element, params) {
+            return  laravelValidation.helpers.parseTime(value,params[0])!=false;
+        },
 
         /**
          * Validate the date is before a given date.
+         * @return {boolean}
          */
-        $.validator.addMethod("laravelBefore", function(value, element, params) {
+        Before: function(value, element, params) {
 
-            var timeCompare=parseFloat(params[0]);
+            var timeCompare=parseFloat(params);
             if (isNaN(timeCompare)) {
-                var el=this.findByName(params[0]);
-                if (el[0]==undefined) {
-                    return false;
-                }
-                timeCompare= helpers.parseTime(this.elementValue(el[0]), el[0]);
-                if ( this.settings.onfocusout ) {
-                    el.unbind( ".validate-laravelBefore" ).bind( "blur.validate-laravelBefore", function() {
-                        $( element ).valid();
-                    });
-                }
+                var target=laravelValidation.helpers.dependentElement(this, element, params);
+                timeCompare= laravelValidation.helpers.parseTime(this.elementValue(target), target);
             }
 
-            var timeValue=helpers.parseTime(value, element);
-            return this.optional(element) || (timeValue !=false && timeValue < timeCompare);
+            var timeValue=laravelValidation.helpers.parseTime(value, element);
+            return  (timeValue !=false && timeValue < timeCompare);
 
-        }, $.validator.format("The :attribute must be a date before {0}."));
+        },
 
         /**
          * Validate the date is after a given date.
+         * @return {boolean}
          */
-        $.validator.addMethod("laravelAfter", function(value, element, params) {
-            var timeCompare=parseFloat(params[0]);
+        After: function(value, element, params) {
+            var timeCompare=parseFloat(params);
             if (isNaN(timeCompare)) {
-                var el=this.findByName(params[0]);
-                if (el[0]==undefined) {
-                    return false;
-                }
-                timeCompare= helpers.parseTime(this.elementValue(el[0]), el[0]);
-                if ( this.settings.onfocusout ) {
-                    el.unbind( ".validate-laravelAfter" ).bind( "blur.validate-laravelAfter", function() {
-                        $( element ).valid();
-                    });
-                }
+                var target=laravelValidation.helpers.dependentElement(this, element, params);
+                timeCompare= laravelValidation.helpers.parseTime(this.elementValue(target), target);
             }
 
-            var timeValue=helpers.parseTime(value, element);
-            return this.optional(element) || (timeValue !=false && timeValue > timeCompare);
+            var timeValue=laravelValidation.helpers.parseTime(value, element);
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Expected '!==' and instead saw '!='.](https://www.codacy.com/app/albert-moreno-t/laravel-jsvalidation/file/842620834/issues/source?bid=2225045&fileBranchId=2805859#l431)